e2e/key/keyprivate/keyprivate.go

```diff
@@ -0,0 +1,136 @@
+
+	for i, line := range lines {
+		if strings.Contains(line, "B") {
+			lines[i] = "P"
+		}
+	}
+	output := strings.Join(lines, "\n")
+	err = ioutil.WriteFile(kpath, []byte(output), 0644)
+	if err != nil {
+		t.Fatalf("Unable to write to file: %v", err)
+	}
+}
+
+func testPrivateKey(t *testing.T) {
+	tests := []struct {
+		name    string
+		args    []string
+		stdin   string
+		file    string
+		corrupt bool
+		succeed bool
+	}{
+		{
+			name:    "export_private",
+			args:    []string{"export", "--secret"},
+			stdin:   "0\n", // TODO: this will need to be '1' at some point in time -> issue #3199
+			file:    defaultKeyFile,
+			succeed: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run("key_run "+tt.name, test.WithoutPrivilege(func(t *testing.T) {
+			os.RemoveAll(filepath.Join(keyPath, defaultKeyFile))
+			out, err := e2e.ExportPrivateKey(t, tt.file)
+			if tt.succeed {
+				if err != nil {
+					t.Log(string(out))
+					t.Fatalf("Unexpected failure: %v", err)
+				}
+
+				t.Run("remove_private_keyring_before_importing", test.WithoutPrivilege(func(t *testing.T) { e2e.RemoveSecretKeyring(t) }))
+				t.Run("import_private_keyring_from:"+tt.name, test.WithoutPrivilege(func(t *testing.T) {
+					b, err := e2e.ImportPrivateKey(t, defaultKeyFile)
+					if err != nil {
+						t.Log(string(b))
+						t.Fatalf("Unable to import key: %v", err)
+					}
+				}))
+			} else {
+				// if the test key is corrupted, try to import it, should fail
+				if tt.corrupt {
+					t.Run("corrupting_key", test.WithoutPrivilege(func(t *testing.T) { corruptKey(t, defaultKeyFile) }))
+					t.Run("import_private_key", test.WithoutPrivilege(func(t *testing.T) {
+						b, err := e2e.ImportKey(t, defaultKeyFile)
+						if err == nil {
+							t.Fatalf("Unexpected success: %s", string(b))
+						}
+					}))
+				} else {
+					if err == nil {
+						t.Log(string(out))
+						t.Fatalf("Unexpected success")
+					}
+				}
+			}
+		}))
+	}
+}
+
+// TestAll is trigered by ../key.go, that is trigered by suite.go in the e3e test directory
+func TestAll(t *testing.T) {
+	err := envconfig.Process("E2E", &testenv)
+	if err != nil {
+		t.Fatal(err.Error())
+	}
+
+	t.Run("backingup_secret_keyring", test.WithoutPrivilege(func(t *testing.T) { e2e.BackupSecretKeyring(t) }))
+	t.Run("importing_test_key", test.WithoutPrivilege(func(t *testing.T) {
+		b, err := e2e.ImportPrivateKey(t, "./key/testdata/e2e_test_key.asc")
+		if err != nil {
+			t.Log(string(b))
+			t.Fatalf("Unable to import test key: %v", err)
+		}
+	}))
+
+	keyPath = testenv.TestDir
+	defaultKeyFile = filepath.Join(keyPath, "exported_private_key")
+
+	// Pull the default public key
+	t.Run("pull_default_key", test.WithoutPrivilege(func(t *testing.T) { e2e.PullDefaultPublicKey(t) }))
+
+	// Run the tests
+	t.Run("public_key", testPrivateKey)
+
+	// Recover the secret keyring
+	t.Run("recovering_secret_keyring", test.WithoutPrivilege(func(t *testing.T) { e2e.RecoverSecretKeyring(t) }))
```

Comment:
this is also not a test, this is teardown.

If you need to set up and tear down for each test, you can wrap the second argument to t.Run with a function that takes a func (t *testing.T) function and returns a func (t *testing.T) function, just like test.WithoutPrivilege does.

For example:

```
func withEmptySecretKeyring(f func (t *testing.T)) {
    return func (t *testing.T) {
        backupSecretKeyring(t)
        defer restoreSecretKeyringt(t)
        f(t)
    }
}
```

`defer` is required becuase defer runs even if the function f panics.